airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/source.py

```diff
@@ -132,7 +132,7 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
         try:
             config = AmazonSellerPartnerConfig.parse_obj(config)  # FIXME: this will be not need after we fix CDK
             stream_kwargs = self._get_stream_kwargs(config)
-            orders_stream = Orders(**stream_kwargs)
+            orders_stream = VendorSalesReports(**stream_kwargs)
```

Comment:
There is a task that says [Source Amazon Seller Partner: not able to connect to Amazon Vendor Central accounts](https://github.com/airbytehq/airbyte/issues/12124). So we need to choose such a report is which supported by both account types

Reply:
Ah! This is just in the `check_connection`, not when actually defining the stream. That makes sense. Thanks!

Reply:
Would you also add "fixed connection check failure when trying to connect to Amazon Vendor Central accounts" or something of the sort to the changelog so that users like that know that their problem is fixed?

Reply:
Sure, I have added it

Reply:
Thanks!

Reply:
Since this change was implemented, we're getting the following error for some of our selling partner sources:

Configuration check failed
HTTPError('400 Client Error: Bad Request for url: https://sellingpartnerapi-na.amazon.com/reports/2021-06-30/reports')

Reply:
Yeah, I don't see anything here that indicates `GET_VENDOR_SALES_REPORT` is a report is good for sellers and vendors:

https://developer-docs.amazon.com/sp-api/docs/report-type-values#vendor-retail-analytics-reports